File: Chapter1_ClassAndObject/Java/Circle.java

public class Circle {
	final double PI = 3.14159;
	
	private Point2D center = new Point2D();
	private double radius;
	
	void set(double x0, double y0, double r) {
		this.center.set(x0, y0);
		this.set(r);
	}
	
	void set(double r) {
		if (r < 0) {
			r = 0;
		}
		
		this.radius = r;
	}
	
	void move(double dx, double dy) {
		center.move(dx, dy);
	}
	
	double area() {
		return PI * this.radius * this.radius;
	}
	
	double perimeter() {
		return 2 * PI * this.radius;
	}
}

File: Chapter1_ClassAndObject/CSharp/Circle.cs
//using System;
//using System.Collections.Generic;
//using System.Linq;
//using System.Text;
//using System.Threading.Tasks;

//namespace Chapter1_ClassAndObject_CSharp
//{
//    class Circle
//    {
//    }
//}

class Circle
{
    const double PI = 3.14159;

    Point2D center = new Point2D();

    double radius;

    public void set(double x0, double y0, double r)
    {
        this.center.set(x0, y0);
        this.set(r);
    }

    public void set(double r)
    {
        //ensure that the radius is nonnegative.
        if(r < 0)
        {
            r = 0;
        }

        this.radius = r;
    }

    public void move(double dx, double dy)
    {
        this.center.move(dx, dy);
    }

    public double area()
    {
        return PI * radius * radius;
    }

    public double perimeter()
    {
        return 2 * PI * radius;
    }
}
